### PR TITLE
feat: shiki on custom `pm` components

### PR DESCRIPTION
**Pull request by cpreston321**
### 🔗 Linked issue

#40 Somwhat related but I wrote a comment on why.

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [X] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

CleanShot 2024-02-01 at 13 53 14@2x
CleanShot 2024-02-01 at 13 53 02@2x

### 📝 Checklist

- [X] I have linked an issue or discussion.
- [X] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by Barbapapazes** [comment]
It's more deeper than this component. Take a look at the deployed documentation, there is not highlight at all.

**Comment by cpreston321** [comment]
@Barbapapazes I understand I made the comment in #40. When I first implemented the components I ran into the same issue where is would only render highlighting in within MDC vs using the component. This was a couple days ago and the highlighting was working fine. Maybe it will get fixed here soon and I will test again once nuxt/content get's updated.

**Comment by cpreston321** [comment]
@Barbapapazes Here is some of the references to the code that was being used: 

 - [ProseCode](https://github.com/nuxt-modules/mdc/blob/main/src/runtime/components/prose/ProseCode.vue)
 - [CodeGroup](https://github.com/nuxt/ui-pro/blob/dev/modules/pro/runtime/components/content/CodeGroup.vue)
 
---

These components don't include highlighting and since @pi0 switched the [`1.guide/index.md`](https://github.com/unjs/docs/blob/main/playground/1.guide/index.md) to use these new components none of the highlighting will work. [Example](https://unjs-docs.pages.dev/guide). There is still issue of the dep issue. Just so you understand the workings of this PR are a little different.

File: layers/core/app/utils/shiki.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { getHighlighter } from 'shiki'
```
Comment (pi0) [inline layers/core/app/utils/shiki.ts:1]: We should use a lazy imported of `shikiji`. And `nuxt/mdc` module already exposes an endpoint for it.

**Comment by pi0** [comment]
Thanks for this PR. I also tried against latest mdc deps but honestly it is something to be fixed in either UI Pro library or content/mdc stack that avoids us to use programmatic tabs. /cc @benjamincanac @farnabaz any help is really appreciated if you had time 🙏🏼 

Adding shiki logic here directly means we need to maintain all the complexity of highlighting (which is not at it's best state)...

I think for now I prefer to not have highlight for npm commands rather than doing this.

**Pull request closed without merging**